vars/promoteRepos.groovy

```diff
@@ -0,0 +1,196 @@
+    println("Repo Version: ${repoVersion}")
+
+    String stagingPkgPathX64 = "${PUBLIC_ARTIFACT_URL}/${jobname}/${revision}/${buildnumber}/linux/x64/${distType}/dist/${filename}/${filename}-${revision}-linux-x64.${distType}"
+    String stagingPkgPathARM64 = "${PUBLIC_ARTIFACT_URL}/${jobname}/${revision}/${buildnumber}/linux/arm64/${distType}/dist/${filename}/${filename}-${revision}-linux-arm64.${distType}"
+
+    String localPath = "${WORKSPACE}/artifacts"
+    String RepoProdPath = "releases/bundle/${filename}/${repoVersion}/${repoType}"
+    String artifactPath = "${localPath}/${RepoProdPath}"
+
+    withCredentials([string(credentialsId: 'jenkins-artifact-promotion-role', variable: 'ARTIFACT_PROMOTION_ROLE_NAME'),
+        string(credentialsId: 'jenkins-aws-production-account', variable: 'AWS_ACCOUNT_ARTIFACT'),
+        string(credentialsId: 'jenkins-artifact-production-bucket-name', variable: 'ARTIFACT_PRODUCTION_BUCKET_NAME')]) {
+            withAWS(role: "${ARTIFACT_PROMOTION_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_ARTIFACT}", duration: 900, roleSessionName: 'jenkins-session') {
+                println("Pulling Prod ${repoType}")
+                sh("aws s3 sync s3://${ARTIFACT_PRODUCTION_BUCKET_NAME}/${RepoProdPath}/ ${artifactPath}/ --no-progress")
+            }
+
+        sh """
+            set -e
+            set +x
+
+            echo "Pulling ${revision} ${distType}"
+            cd ${artifactPath}
+            curl -SLO ${stagingPkgPathX64}
+            curl -SLO ${stagingPkgPathARM64}
+
+            ls -l
+        """
+
+        if (repoType.equals("yum")) {
+
+            println("Yum Repo Starts")
+
+            sh """
+                set -e
+                set +x
+
+                cd ${artifactPath}
+                rm -vf repodata/repomd.xml.asc
+    
+                echo "Update repo metadata"
+                createrepo --update .
+    
+                # Rename .xml to .pom for signing
+                # Please do not add .xml to signer filter
+                # As maven have many .xml and we do not want to sign them
+                # This is an outlier case for yum repo only
+                mv -v repodata/repomd.xml repodata/repomd.pom
+    
+                echo "Complete metadata update, awaiting signing repomd.xml"
+                cd -
+            """
+
+            signArtifacts(
+                artifactPath: "${artifactPath}/repodata/repomd.pom",
+                sigtype: '.sig',
+                platform: 'linux'
+            )
+
+            sh """
+                set -e
+                set +x
+    
+                cd ${artifactPath}/repodata/
+    
+                ls -l
+    
+                mv -v repomd.pom repomd.xml
+                mv -v repomd.pom.sig repomd.xml.sig
+    
+                # This step is required as yum only accept .asc and signing workflow only support .sig
+                cat repomd.xml.sig | gpg --enarmor | sed 's@ARMORED FILE@SIGNATURE@g' > repomd.xml.asc
```

Comment:
You can use `.asc` in previous command, We have the fix in that actually creates an `.asc` signature now

Reply:
I know, but this is to completely restore the exact command of previous iteration, we can always improve this later, I just do not want to change the behavior now.

Reply:
We can test this in this iteration. Seems alot hacky to it this way when signer is already taking care of it.

Reply:
fixed now.